dts/bindings/serial/st,stm32-uart.yaml

```diff
@@ -11,6 +11,12 @@ properties:
     interrupts:
       required: true
 
+    single-wire:
+      type: boolean
+      required: false
+      description: |
+        Enable the single wire functionality
```

Comment:
Please provide more details here.
I suggest (just a proposal, don't hesitate to fix based on you experience of the feature)
```
Enables single-wire half-duplex communication. 
Using this mode, TX and RX lines are internally connected, so only TX or RX pin should be configured.
RX/TX conflicts should be handled on user side.
```

Reply:
Sounds good. Never seen the option to choose using TX or RX pin, F1, F3 and H7 always use external TX pin, so adapted in that direction.